src/instructlab/utils.py

```diff
@@ -286,15 +292,23 @@ def _validate_documents(
             opened_files = False
             for pattern in file_patterns:
                 for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
-                    if os.path.isfile(file_path) and file_path.endswith(".md"):
-                        with open(file_path, "r", encoding="utf-8"):
-                            # Success! we could open the file.
-                            # We don't actually care about the contents, though.
-                            opened_files = True
+                    logger.debug("Checking file: %s", file_path)
+                    if os.path.isfile(file_path):
+                        if file_path.endswith(".md"):
+                            # Open markdown files in text mode
+                            with open(file_path, "r", encoding="utf-8"):
+                                # Successfully opened the markdown file.
+                                opened_files = True
+                        elif file_path.endswith(".pdf"):
```

Comment:
to be more future facing it may be better to define a slice of accepted formats and loop through it rather than add new if-else conditions for each new file format IMO 
that way extending to new formats will be easier and we don't repeat code

Reply:
Agreed on this point